homeassistant/components/climate/evohome.py

```diff
@@ -279,93 +566,60 @@ def set_operation_mode(self, operation_mode):
         Currently limited to 'Auto', 'AutoWithEco' & 'HeatingOff'. If 'Away'
         mode is needed, it can be enabled via turn_away_mode_on method.
         """
-        self._set_operation_mode(HA_STATE_TO_EVO.get(operation_mode))
+        self._set_operation_mode(HA_STATE_TO_TCS.get(operation_mode))
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True when the evohome Controller should be polled.
+
+        The evohome Controller will be polled once every 300 seconds by
+        default.
+        """
+        timeout = datetime.now() + timedelta(seconds=55)
```

Comment:
I am pretty certain that it is checked every minute (averages 61 secs on my system).  If `True`, HA calls `self.update()`, otherwise it does not.

In fact, my code wouldn't be working unless it did!  Here is an example - I added a `_LOGGER.warn()` to show it working:
```
2018-11-13 22:22:36 INFO (MainThread) [homeassistant.core] Timer:starting
2018-11-13 22:23:36 WARNING (MainThread) [homeassistant.components.climate.evohome] should_poll(3432522)=False
2018-11-13 22:24:37 WARNING (MainThread) [homeassistant.components.climate.evohome] should_poll(3432522)=False
2018-11-13 22:25:37 WARNING (MainThread) [homeassistant.components.climate.evohome] should_poll(3432522)=False
2018-11-13 22:26:38 WARNING (MainThread) [homeassistant.components.climate.evohome] should_poll(3432522)=False
2018-11-13 22:27:39 WARNING (MainThread) [homeassistant.components.climate.evohome] should_poll(3432522)=True
2018-11-13 22:27:39 DEBUG (SyncWorker_3) [homeassistant.components.climate.evohome] _update_state_data(): API call [1 request(s)]: client.locations[0].status()...
   ...
```

HA has always done this as far as I recall - since about 0.77, certainly.

Reply:
If it would be false the first time, it would never check again.

Reply:
Here's the code:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_platform.py#L228-L231

So if no entities of the platform should poll when they are added the periodic update won't start.

Reply:
Thanks for explaining the distinction - I wasn't aware of the edge case.

In any case, the component wont ever have a problem with this, as there will _always_ be one (and only one) Controller, and during it's `__init__()`, there is:
```python
self._timers['statusUpdated'] = datetime.min
```
... which will _always_ cause`should_poll()` to be `True` the first time its called:
```python
def should_poll(self) -> bool:
    timeout = datetime.now() + timedelta(seconds=55)
    expired = timeout > self._timers['statusUpdated'] + \
        timedelta(seconds=self._params[CONF_SCAN_INTERVAL])
    return expired  # if expired then self.update()
```

I think having this login in `should_poll()` is more elegant, and keeps the Controller's `update()` as simple as possible.

Are you happy to leave it as is?

Reply:
Also, I am changing from `load_platform()` / `add_entities` to their `async` counterparts...  Of coure, there is no I/O in `async_setup_platform`.

Reply:
I think it probably would be more clear if the component schedules its own update interval and we don't rely on the core polling mechanism in this case. That's what we normally do when we need a custom interval behavior.

So set `should_poll` to `False` and schedule a periodic update in the component that signals the entitites via our dispatch helper.

Reply:
What about the `@Throttle(MIN_TIME_BETWEEN_UPDATES)` as used by `climate/netatmo.py`?

Reply:
What about it?

Reply:
Yeah, sorry, I didn't realize it was a simple throttle. If it's ok to throttle per entity, then you can use the Throttle decorator like that. :+1:

Reply:
I have switched back to the old way for now - I couldn't work out to have a variable-length `Throttle`.

`should_poll` is always `True` for the Controller (hub/parent), and `False` for its children.  The scan interval logic is in `update()`.